cmd/snap/cmd_snapshot.go

```diff
@@ -0,0 +1,297 @@
+		if err == noWait {
+			return nil
+		}
+		return err
+	}
+
+	y := &savedCmd{
+		clientMixin:   x.clientMixin,
+		durationMixin: x.durationMixin,
+		ID:            snapshotID(setID),
+	}
+	return y.Execute(nil)
+}
+
+type forgetCmd struct {
+	waitMixin
+	Positional struct {
+		ID    snapshotID          `positional-arg-name:"<id>"`
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+var shortForgetHelp = i18n.G("Delete a snapshot")
+var longForgetHelp = i18n.G(`
+The forget command deletes a snapshot.
+`)
+
+func (x *forgetCmd) Execute([]string) error {
+	snaps := installedSnapNames(x.Positional.Snaps)
+	changeID, err := x.client.ForgetSnapshots(uint64(x.Positional.ID), snaps)
+	if err != nil {
+		return err
+	}
+	_, err = x.wait(changeID)
+	if err == noWait {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if len(snaps) > 0 {
+		// TRANSLATORS: the %s is a comma-separated list of quoted snap names
+		fmt.Fprintf(Stdout, i18n.NG("Snapshot #%d of snap %s forgotten.\n", "Snapshot #%d of snaps %s forgotten.\n", len(snaps)), x.Positional.ID, strutil.Quoted(snaps))
+	} else {
+		fmt.Fprintf(Stdout, i18n.G("Snapshot #%d forgotten.\n"), x.Positional.ID)
+	}
+	return nil
+}
+
+type checkSnapshotCmd struct {
+	waitMixin
+	Users      string `long:"users"`
+	Positional struct {
+		ID    snapshotID          `positional-arg-name:"<id>"`
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+var shortCheckHelp = i18n.G("Check a snapshot")
+var longCheckHelp = i18n.G(`
+The check-snapshot command checks a snapshot against its hashsums.
+`)
+
+func (x *checkSnapshotCmd) Execute([]string) error {
+	snaps := installedSnapNames(x.Positional.Snaps)
+	var users []string
+	if len(x.Users) > 0 {
+		users = strings.Split(x.Users, ",")
+	}
+	changeID, err := x.client.CheckSnapshots(uint64(x.Positional.ID), snaps, users)
+	if err != nil {
+		return err
+	}
+	_, err = x.wait(changeID)
+	if err == noWait {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// TODO: also mention the home archives that were actually checked
+	if len(snaps) > 0 {
+		// TRANSLATORS: the %s is a comma-separated list of quoted snap names
+		fmt.Fprintf(Stdout, i18n.G("Snapshot #%d of snaps %s check passed.\n"),
+			x.Positional.ID, strutil.Quoted(snaps))
+	} else {
+		fmt.Fprintf(Stdout, i18n.G("Snapshot #%d check passed.\n"), x.Positional.ID)
+	}
+	return nil
+}
+
+type restoreCmd struct {
+	waitMixin
+	Users      string `long:"users"`
+	Positional struct {
+		ID    snapshotID          `positional-arg-name:"<id>"`
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+var shortRestoreHelp = i18n.G("Restore a snapshot")
+var longRestoreHelp = i18n.G(`
+The restore command restores a snapshot.
+`)
+
+func (x *restoreCmd) Execute([]string) error {
+	snaps := installedSnapNames(x.Positional.Snaps)
+	var users []string
```

Comment:
Perhaps a small helper for handling the users field across many commands:
```
func splitUsers(users string) []string {
   if len(users) > 0 {
     return strings.Split(users, ",")
   }
   return nil
}
```

Reply:
I'll move `daemon.splitQS` to `strutil`

Reply:
(... in a followup)